### ISSUE TITLE
[MNG-4441] Profile Activation Regression in Maven3

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#5717]
**[Markus Heberling](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=tisoft_media)** opened **[MNG-4441](https://issues.apache.org/jira/browse/MNG-4441?redirect=false)** and commented

As written here http://www.sonatype.com/people/2009/11/maven-30-alpha-3-released/#IDComment43056417 I have a problem with profile activation in Maven3.

I have a profile that activates if target/generated-sources/castor is missing and includes castor in the build process that will create that directory and generate the sources into it.

If the directory exists and I do mvn clean install, the build fails with maven 3. Maven does not activate the profile since the directory exists at the beginning. But the clean step will delete it of course and so the sources are missing and the build fails. With maven 2 this works.

---

**Affects:** 3.0-alpha-3

**Attachments:**
- [pom.xml](https://issues.apache.org/jira/secure/attachment/12712334/pom.xml) (_1.96 kB_)

**Comment by jira-importer** [comment]
**[Benjamin Bentmann](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bentmann)** commented

Any chance you can make a reproducible test project for this?

**Comment by jira-importer** [comment]
**[Benjamin Bentmann](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bentmann)** commented

Based on the little information given, I created this example project. This works the same with Maven 2 and 3, i.e. the profile is not activated when the output directory exists before the invocation of "mvn clean generate-sources".

I also checked the Maven 2 sources and didn't see any spots that would reevaluate profiles in the middle of the build. In other words, both Maven 2 and Maven 3 evaluate profiles only once, namely before any build operation is performed.
